config/7.1.0/securebanking/ig/scripts/groovy/RepoConsent.groovy

```diff
@@ -46,46 +47,43 @@ def buildPatchRequest(incomingRequest, intentType) {
         ]);
     }
 
-    // Domestic Payment Intent only
-    /*
-    schemeName: "UK.OBIE.SortCodeAccountNumber",
-            identification: "79126738233670",
-            name: "7b78b560-6057-41c5-bf1f-1ed590b1c30b",
-            secondaryIdentification:
-     */
+    // Payments Intent: Update Debtor account and accountId
     if (incomingRequest.data && incomingRequest.data.debtorAccount) {
-        if(incomingRequest.data.debtorAccount.schemeName) {
-            body.push([
-
-                    "operation": "add",
-                    "field"    : "OBIntentObject/Data/Initiation/DebtorAccount/SchemeName",
-                    "value"    : incomingRequest.data.debtorAccount.schemeName
-            ])
-        }
-        if(incomingRequest.data.debtorAccount.identification) {
-            body.push([
-
-                    "operation": "add",
-                    "field"    : "OBIntentObject/Data/Initiation/DebtorAccount/Identification",
-                    "value"    : incomingRequest.data.debtorAccount.identification
-            ])
+        if(incomingRequest.data.debtorAccount.accounts[0]) {
```

Comment:
is there a use case for debtor account containing an array?

Reply:
Now the `debtorAccount` object is a `FRFinancialAccount` instead of a `FRAccountIdentifier` to update properly the accountId, and then we get the first account from the list that will contains only one account, the account selected by the PSU user when the decision submitted is `Authorised`.